src/paas_charm/django/charm.py

```diff
@@ -75,6 +75,20 @@ def get_cos_dir(self) -> str:
         """
         return str((pathlib.Path(__file__).parent / "cos").absolute())
 
+    def get_framework_config(self) -> BaseModel:
+        """Return the framework related configurations.
+
+        The method is overridden to inject the base_url, that can be an ingress URL or a k8s svc
+        url, to the list of allowed hosts.
+
+        Returns:
+             Framework related configurations.
+        """
+        base_model = super().get_framework_config()
+        # base_model can be downcasted to a DjangoConfig, and allowed_hosts is really a list.
+        base_model.allowed_hosts.append(self._base_url)  # type: ignore
```

Comment:
I'm a bit out of the loop, so this might be a random question: what if the ingress URL changes?

Reply:
The point is good, we need to make sure we reload the app if the ingress changes and make sure that we reload the allowed hosts

Reply:
As `paas-charm` is designed in a holistic way, every event that changes something will call a replan, and as the env vars change, the service will be restarted. As there is no harm in an extra test I have added a ~~"unit"~~ test for it.

(Besides I have fixed a bug that I did not see because of the '*', I updated one integ test so it fails if the allowed_host is not correct).